packages/api/src/controllers/nominationController.js

```diff
@@ -0,0 +1,163 @@
+          .select('workflow_id')
+          .where('status', initialStatus)
+          .where('type_group', data.nomination_type)
+          .first();
+        data.workflow_id = workflow_id;
+        // Add nomination
+        const nomination = await baseController.postWithResponse(NominationModel, data, req, {
+          trx,
+        });
+        data.nomination_id = nomination.nomination_id;
+        // Add status change entry
+        const status = await baseController.postWithResponse(NominationStatusModel, data, req, {
+          trx,
+        });
+        const result = { nomination, status };
+        await trx.commit();
+        res.status(201).send(result);
+      } catch (error) {
+        console.log(error);
+        let violationError = false;
+        if (error instanceof UniqueViolationError) {
+          violationError = true;
+          await trx.rollback();
+          res.status(400).json({
+            error,
+            violationError,
+          });
+        } else {
+          await trx.rollback();
+          res.status(400).json({
+            error,
+            violationError,
+          });
+        }
+      }
+    };
+  },
+
+  /**
+   * This updates a single nomination based on a nomination id and nomination type in body.
+   * @returns The updated nomination row.
+   */
+  updateNomination() {
+    return async (req, res) => {
+      const trx = await transaction.start(Model.knex());
+      try {
+        const data = req.body;
+        const params = req.params;
+        //overwrite verified farm_id
+        data.farm_id = req.headers.farm_id;
+        const nomination = await baseController.put(
+          NominationModel,
+          params.nomination_id,
+          data,
+          req,
+          {
+            trx,
+          },
+        );
+        const result = { nomination };
+        await trx.commit(result);
+        res.status(201).send(result);
+      } catch (error) {
+        console.log(error);
+        let violationError = false;
+        if (error instanceof UniqueViolationError) {
+          violationError = true;
+          await trx.rollback();
+          res.status(400).json({
+            error,
+            violationError,
+          });
+        }
+        //handle more exceptions
+        else {
+          await trx.rollback();
+          res.status(400).json({
+            error,
+            violationError,
+          });
+        }
+      }
+    };
+  },
+
+  /**
+   * Soft deletes a nomination based on nomination id
+   * @returns Boolean success
+   */
+  deleteNomination() {
+    return async (req, res) => {
+      const trx = await transaction.start(Model.knex());
+      try {
+        const params = req.params;
+        const success = await baseController.delete(NominationModel, params.nomination_id, req, {
+          trx,
+        });
+        await trx.commit();
+        res.status(201).send({ success });
```

Comment:
I found that other delete APIs I referred were returning a 200 status code like below:
`res.sendStatus(200);`

Can we confirm what the response should be?
Other than that, everything worked great! Great job!!

Reply:
200 Kike had a confluence doc on status code :)

Reply:
Cool found it in "The api package", he references REST standards. Thanks for finding I complete forgot about this part. I also fixed the update response codes in update as well as delete based on the other controllers.